client/src/lib/types.ts

```diff
@@ -26,6 +26,16 @@ export type Photo = {
 
 export type User = {
   username: string | undefined;
-  token: string | undefined;
   userId: number | undefined;
 };
+
+export type Auth = {
+  user: User;
+  token: string;
+};
+
+export type UserContextValues = {
```

Comment:
To me, this type is more naturally owned by `UserContext.tsx`, so I would put it there. Pretty much everywhere you use `UserContextValues` you are also going to use `UserContext`, so you'd just put them both into the same import.

I find a file named `types.ts` to be an anti-pattern -- something to avoid. I think types are more naturally put next to the function that "owns" them.

There are some types that don't have a natural owner or that are used ubiquitously. Auth and User would probably fall into that category. Basically, types that define your database types fall into that category.